client/landing/stepper/declarative-flow/import-flow.ts

```diff
@@ -0,0 +1,157 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { Design, isBlankCanvasDesign } from '@automattic/design-picker';
+import { IMPORT_FOCUSED_FLOW } from '@automattic/onboarding';
+import { useViewportMatch } from '@wordpress/compose';
+import { useSelect } from '@wordpress/data';
+import { ImporterMainPlatform } from 'calypso/blocks/import/types';
+import { useQuery } from 'calypso/landing/stepper/hooks/use-query';
+import { useSiteSlugParam } from 'calypso/landing/stepper/hooks/use-site-slug-param';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import { StepPath } from './internals/steps-repository';
+import { Flow, ProvidedDependencies } from './internals/types';
+
+export const importFlow: Flow = {
+	name: IMPORT_FOCUSED_FLOW,
+
+	useSteps() {
+		return [
+			'import',
+			'importList',
+			'importReady',
+			'importReadyNot',
+			'importReadyWpcom',
+			'importReadyPreview',
+			'importerWix',
+			'importerBlogger',
+			'importerMedium',
+			'importerSquarespace',
+			'importerWordpress',
+			'designSetup',
+			'patternAssembler',
+			'processing',
+		] as StepPath[];
+	},
+
+	useStepNavigation( _currentStep, navigate ) {
+		const urlQueryParams = useQuery();
+		const siteSlugParam = useSiteSlugParam();
+		const isDesktop = useViewportMatch( 'large' );
+		const selectedDesign = useSelect( ( select ) => select( ONBOARD_STORE ).getSelectedDesign() );
+
+		const appendFlowQueryParam = ( path: string ) => {
+			const joinChar = path.includes( '?' ) ? '&' : '?';
+
+			return `${ path }${ joinChar }flow=${ this.name }`;
+		};
+
+		const exitFlow = ( to: string ) => {
```

Comment:
I took a quick look and looks like if we trigger `exitFlow` and not calling [setPendingAction](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/site-setup-flow.ts#L126) it'll navigating to `processing` and for some reason it'll trigger `exitFlow` again with homepage redirection. So maybe we still need to have `setPendingAction` here to prevent the extra call to `exitFlow`. I hope this gives you enough information to start checking with it. Please let me know if you need more details on this, thanks.

Reply:
Thanks, man!

Exactly, adding `setPendingAction` solves it.